ecgtools/core.py

```diff
@@ -213,21 +216,125 @@ def parse_files_attributes(
+        format_column : str
+            The column name which contains the data format, allowing for multiple data assets
+            (file formats) types in one catalog. Mutually exclusive with `data_format`.
+        groupby_attrs : list
+            Column names (attributes) that define data sets that can be aggegrated,
+            default None
+        aggregations : List[dict]
+            List of aggregations to apply to query results, default None
+        esmcat_version : str
+            The ESM Catalog version the collection implements, default None
+        cat_id : str
+            Identifier for the collection, default None
+        description : str
+            Detailed multi-line description to fully explain the collection,
+            default None
+        attributes : List[dict]
+            A list of attributes. An attribute dictionary describes a column
+            in the catalog CSV file.
+        kwargs : Additional keyword arguments are passed through to the
+                 :py:class:`~pandas.DataFrame.to_csv` method.
 
         Returns
         -------
         `ecgtools.Builder`
 
+        Notes
+        -----
+        See https://github.com/NCAR/esm-collection-spec/blob/master/collection-spec/collection-spec.md
+        for more details on ESM collection specification.
+
         """
+
         self.df = pd.DataFrame(self.entries)
+
+        esmcol_data = OrderedDict()
+        if esmcat_version is None:
+            esmcat_version = '0.0.1'
+
+        if cat_id is None:
+            cat_id = ''
+
+        if description is None:
+            description = ''
+        esmcol_data['esmcat_version'] = esmcat_version
+        esmcol_data['id'] = cat_id
+        esmcol_data['description'] = description
+        esmcol_data['catalog_file'] = catalog_file
+        if attributes is None:
+            attributes = []
+            for column in self.df.columns:
+                attributes.append({'column_name': column, 'vocabulary': ''})
+
+        esmcol_data['attributes'] = attributes
+        esmcol_data['assets'] = {'column_name': path_column}
+        if (data_format is not None) and (format_column is not None):
+            raise ValueError('data_format and format_column are mutually exclusive')
+
+        if data_format is not None:
+            esmcol_data['assets']['format'] = data_format
+        else:
+            esmcol_data['assets']['format_column_name'] = format_column
+        if groupby_attrs is None:
+            groupby_attrs = [path_column]
+
+        if aggregations is None:
+            aggregations = []
+        esmcol_data['aggregation_control'] = {
+            'variable_column_name': variable_column,
+            'groupby_attrs': groupby_attrs,
+            'aggregations': aggregations,
+        }
+
+        json_path = f'{catalog_file.split(".")[0]}.json'
+
+        with open(json_path, mode='w') as outfile:
+            json.dump(esmcol_data, outfile, indent=2)
```

Comment:
Should the `esmcol_data` object be made public?  There is a public version of the CSV file in the `df` attribute, so maybe a public version of the JSON file should be made public?

See my notes in the review summary.